ci/travis_set_build.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Update the version string with a build number"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import io
+import os
+import re
+import sys
+
+
+PROJECT_ROOT = os.path.abspath(os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    os.pardir)
+)
+BUILD_NUMBER = os.environ.get('TRAVIS_BUILD_NUMBER', '')
+VERSION_FILE_PATHS = ('src', 'watchmaker', '__init__.py')
+
+
+def replace(file_path, pattern, repl, flags=0):
+    with io.open(file_path, mode="r+") as fh_:
+        file_contents = fh_.read()
+        file_contents = re.sub(pattern, repl, file_contents, flags=flags)
+        fh_.seek(0)
+        fh_.truncate()
+        fh_.write(file_contents)
+
+
+def main(args):
+    skip = args.skip
+    build = args.build
+    file_paths = args.file_paths
+
+    if skip:
+        print(
+            'Not updating version for this build, `skip` set to "{0}"'
+            .format(skip)
+        )
+        sys.exit(0)
+
+    # The version line must have the form
+    # __version__ = 'ver'
+    pattern = r"^(__version__ = ['\"])([^'\"]*)(['\"])"
+    repl = r"\g<1>\g<2>.dev{0}\g<3>".format(build)
+    version_file = os.path.join(PROJECT_ROOT, *file_paths)
+    print(
+        'Updating version in version_file "{0}" with build "{1}"'
+        .format(version_file, build)
+    )
+    replace(version_file, pattern, repl, flags=re.M)
+
```

Comment:
Instead of using a one sided conditional to sys.exit the middle of a small main, why not break out the things into a function and use else to call it. This will also remove the dep on sys.

```

def append_file(build, file_paths):
    # The version line must have the form
    # __version__ = 'ver'
    pattern = r"^(__version__ = ['\"])([^'\"]*)(['\"])"
    repl = r"\g<1>\g<2>.dev{0}\g<3>".format(build)
    version_file = os.path.join(PROJECT_ROOT, *file_paths)
    print(
        'Updating version in version_file "{0}" with build "{1}"'
        .format(version_file, build)
    )
    replace(version_file, pattern, repl, flags=re.M)

def main(args):
    skip = args.skip
    build = args.build
    file_paths = args.file_paths

    if skip:
        print(
            'Not updating version for this build, `skip` set to "{0}"'
            .format(skip)
        )
    else:
        append_file(build, file_paths)
```